services/databaseEncryptionService.ts

```diff
@@ -0,0 +1,266 @@
+  
+  return error.message.includes('cancel');
+}
+
+function classifyError(error: unknown): EncryptionError {
+  if (error instanceof EncryptionError) {
+    return error;
+  }
+
+  if (isCancellationError(error)) {
+    return new EncryptionError(ERROR_CODES.USER_CANCELLED, 'Authentication was cancelled');
+  }
+
+  const message = error instanceof Error ? error.message : String(error);
+  
+  if (message.includes('Authentication is already in progress')) {
+    return new EncryptionError(ERROR_CODES.AUTH_IN_PROGRESS, 'Authentication is in progress. Please try again.');
+  }
+
+  if (message.includes('not available') || message.includes('SecureStore')) {
+    console.error('[Encryption] SecureStore unavailable:', error);
+    return new EncryptionError(ERROR_CODES.SECURE_STORE_UNAVAILABLE, 'Secure storage is not available on this device');
+  }
+
+  console.error('[Encryption] Unexpected error:', error);
+  return new EncryptionError(ERROR_CODES.SECURE_STORE_UNAVAILABLE, 'Failed to initialize encryption - secure storage may be unavailable');
+}
+
+async function createNewKey(): Promise<Uint8Array> {
+  const key = generateRandomKey();
+  await SecureStore.setItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, base64.fromByteArray(key), { requireAuthentication: false });
+  await setStoredEncryptionMode('basic');
+  return key;
+}
+
+async function loadExistingKey(): Promise<Uint8Array> {
+  console.log('[EncryptionService] loadExistingKey called');
+  const mode = await getStoredEncryptionMode();
+  console.log('[EncryptionService] Current encryption mode:', mode);
+  console.log('[EncryptionService] Will require authentication:', mode === 'protected');
+  
+  if (mode === 'protected') {
+    console.log('[EncryptionService] 🔐 BIOMETRIC PROMPT - About to trigger for key retrieval');
+  }
+  
+  const keyBase64 = await SecureStore.getItemAsync(SECURE_STORE_KEYS.ENCRYPTION_KEY, { requireAuthentication: mode === 'protected' });
+  
+  if (mode === 'protected') {
+    console.log('[EncryptionService] ✅ BIOMETRIC PROMPT - Completed successfully');
+  }
+  
+  if (!keyBase64) {
+    console.error('[Encryption] Key missing');
+    throw new EncryptionError(ERROR_CODES.KEY_CORRUPTION, 'Encryption key is missing. Your data cannot be decrypted.');
+  }
+
+  console.log('[EncryptionService] Key loaded successfully');
+  return base64.toByteArray(keyBase64);
+}
+
+export async function initializeEncryption(): Promise<void> {
+  console.log('[EncryptionService] initializeEncryption called', {
+    hasKeyCache: !!keyCache,
+    hasInitPromise: !!initPromise,
+    timestamp: new Date().toISOString(),
+  });
+
+  if (keyCache) {
+    console.log('[EncryptionService] Using cached key, skipping initialization');
+    return;
+  }
+  
+  if (initPromise) {
+    console.log('[EncryptionService] Initialization already in progress, waiting...');
+    return initPromise;
+  }
+
+  initPromise = (async () => {
```

Comment:
What's the point of wrapping all of this in this big lambda that gets immediately executed?